pmd-test/src/main/java/net/sourceforge/pmd/testframework/SimpleAggregatorTst.java

```diff
@@ -20,13 +20,20 @@
     private List<Rule> rules = new ArrayList<>();
 
     /**
-     * Configure the rule tests to be executed. Implement this method in
-     * subclasses by calling adRule.
+     * Configure the rule tests to be executed. Override this method in
+     * subclasses by calling addRule.
+     * <p>The default implementation will use the package name and test class name,
+     * to determine the ruleset and rule under test.
      *
      * @see #addRule(String, String)
      */
     protected void setUp() {
-        // empty, to be overridden
+        String[] packages = getClass().getPackage().getName().split("\\.");
+        String categoryName = packages[packages.length - 1];
```

Comment:
This is PMD specific, I guess it would probably fail most of the time outside of PMD. Wouldn't it be better to use our own subclass of SimpleAggregatorTst (or RuleTst, since we test a single rule)?

Reply:
That's a good point. The default implementation will probably only work for PMD itself, hence in the documentation this method is always overridden, if you create your own tests.
I'll see, what I can do about this...